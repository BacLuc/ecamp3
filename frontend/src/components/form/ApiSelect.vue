<!--
Displays a field as a v-select + write access via API wrapper
-->

<template>
  <api-wrapper
    v-bind="$props">
    <template slot-scope="wrapper">
      <v-select
        :value="wrapper.localValue"
        v-bind="$attrs"
        :readonly="readonly"
        :disabled="disabled"
        :error-messages="wrapper.errorMessages"
        hide-details="auto"
        outlined
        @input="wrapper.on.input"
        @blur="wrapper.on.touch">
        <status-icon slot="append" :status="wrapper.status" />
      </v-select>
    </template>
  </api-wrapper>
</template>

<script>
import { apiPropsMixin } from '@/mixins/apiPropsMixin'
import ApiWrapper from './ApiWrapper'
import StatusIcon from './StatusIcon'

export default {
  name: 'ApiSelect',
  components: { ApiWrapper, StatusIcon },
  mixins: [apiPropsMixin],
  props: {
    // disable delay per default
    autoSaveDelay: { type: Number, default: 0, required: false }
  },
  data () {
    return {
    }
  }
}
</script>

<style scoped>
</style>
