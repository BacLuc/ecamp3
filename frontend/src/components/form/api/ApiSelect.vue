<!--
Displays a field as a e-select + write access via API wrapper
-->

<template>
  <api-wrapper
    v-slot="wrapper"
    :relation="relation"
    v-bind="$props"
    v-on="$listeners">
    <e-select
      :value="wrapper.localValue"
      v-bind="$attrs"
      :readonly="wrapper.readonly"
      :disabled="disabled"
      :error-messages="wrapper.errorMessages"
      :loading="wrapper.isSaving || wrapper.isLoading ? 'secondary' : false"
      :outlined="outlined"
      :filled="filled"
      :dense="dense"
      @input="wrapper.on.input">
      <template #append>
        <v-icon>mdi-menu-down</v-icon>
        <api-wrapper-append :wrapper="wrapper" />
      </template>
    </e-select>
  </api-wrapper>
</template>

<script>
import { apiPropsMixin } from '@/mixins/apiPropsMixin'
import ApiWrapper from './ApiWrapper'
import ApiWrapperAppend from './ApiWrapperAppend'

export default {
  name: 'ApiSelect',
  components: { ApiWrapper, ApiWrapperAppend },
  mixins: [apiPropsMixin],
  props: {
    // disable delay per default
    autoSaveDelay: { type: Number, default: 0, required: false },
    relation: { type: String, default: '' }
  },
  data () {
    return {
    }
  }
}
</script>

<style scoped>
</style>
