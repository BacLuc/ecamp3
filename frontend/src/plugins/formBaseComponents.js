// Global registration of all Vue components in folder components/form/base

// import upperFirst from 'lodash/upperFirst'
// import camelCase from 'lodash/camelCase'

import ECheckbox from '@/components/form/base/ECheckbox.vue'
import EColorPicker from '@/components/form/base/EColorPicker.vue'
import EDatePicker from '@/components/form/base/EDatePicker.vue'
import ERichtext from '@/components/form/base/ERichtext.vue'
import ESelect from '@/components/form/base/ESelect.vue'
import ESwitch from '@/components/form/base/ESwitch.vue'
import ETextarea from '@/components/form/base/ETextarea.vue'
import ETextField from '@/components/form/base/ETextField.vue'
import ETimePicker from '@/components/form/base/ETimePicker.vue'

/*
const requireComponent = require.context(
  // The relative path of the components folder
  '../components/form/base',
  // Whether or not to look in subfolders
  false,
  // The regular expression used to match base component filenames
  /[A-Z]\w+\.(vue|js)$/
) */

class FormBaseComponentsPlugin {
  install (Vue, options) {
    Vue.component('ECheckbox', ECheckbox)
    Vue.component('EColorPicker', EColorPicker)
    Vue.component('EDatePicker', EDatePicker)
    Vue.component('ERichtext', ERichtext)
    Vue.component('ESelect', ESelect)
    Vue.component('ESwitch', ESwitch)
    Vue.component('ETextarea', ETextarea)
    Vue.component('ETextField', ETextField)
    Vue.component('ETimePicker', ETimePicker)

    /*
    requireComponent.keys().forEach(fileName => {
      // Get component config
      const componentConfig = requireComponent(fileName)

      // Get PascalCase name of component
      const componentName = upperFirst(
        camelCase(
          // Gets the file name regardless of folder depth
          fileName
            .split('/')
            .pop()
            .replace(/\.\w+$/, '')
        )
      )

      // Register component globally
      Vue.component(
        componentName,
        // Look for the component options on `.default`, which will
        // exist if the component was exported with `export default`,
        // otherwise fall back to module's root.
        componentConfig.default || componentConfig
      )
    }) */
  }
}

export default new FormBaseComponentsPlugin()
