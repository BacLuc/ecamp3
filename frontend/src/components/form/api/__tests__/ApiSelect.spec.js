// Libraries
import Vue from 'vue'
import Vuetify from 'vuetify'

import formBaseComponents from '@/plugins/formBaseComponents'

import { mount } from '@vue/test-utils'
import ApiSelect from '../ApiSelect.vue'

Vue.use(Vuetify)
Vue.use(formBaseComponents)

describe('ApiTextField.vue', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  // keep this the first test --> otherwise element IDs change constantly
  test('renders correctly', () => {
    const props = {
      value: 'Test Value',

      /* field name and URI for saving back to API */
      fieldname: 'test-field',
      uri: 'test-field/123',

      /* display label */
      label: 'Test Field',

      /* overrideDirty=true will reset the input if 'value' changes, even if the input is dirty. Use with caution. */
      overrideDirty: false,

      /* enable/disable auto save */
      autoSave: true,

      /* Validation criteria */
      required: true
    }

    const wrapper = mount(ApiSelect, {
      vuetify,
      propsData: props
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
