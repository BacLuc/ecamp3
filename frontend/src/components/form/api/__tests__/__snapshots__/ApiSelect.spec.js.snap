// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiTextField.vue renders correctly 1`] = `
<form
  class="v-form e-form-container"
  novalidate="novalidate"
>
  <div
    class="e-form-container"
    outlined="true"
    required="required"
    value="Test Value"
  >
    <div
      class="v-input v-input--hide-details theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select"
    >
      <div
        class="v-input__control"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="list-11"
          class="v-input__slot"
          role="button"
        >
          <fieldset
            aria-hidden="true"
          >
            <legend
              style="width: 0px;"
            >
              <span>
                â€‹
              </span>
            </legend>
          </fieldset>
          <div
            class="v-select__slot"
          >
            <label
              class="v-label theme--light"
              for="input-11"
              style="left: 0px; position: absolute;"
            >
              Test Field
            </label>
            <div
              class="v-select__selections"
            >
              <input
                aria-readonly="false"
                autocomplete="off"
                id="input-11"
                readonly="readonly"
                required="required"
                type="text"
              />
            </div>
            <div
              class="v-input__append-inner"
            >
              <i
                aria-hidden="true"
                class="v-icon notranslate mdi mdi-menu-down theme--light"
              />
               
              <div
                class="d-flex"
                style="margin-top: -5px;"
              >
                <div
                  class="checkIconContainer"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate checkIcon mdi mdi-content-save theme--light green--text"
                  />
                </div>
                 
                <!---->
                 
                <!---->
              </div>
            </div>
            <input
              type="hidden"
              value="Test Value"
            />
          </div>
          <div
            class="v-menu"
          >
            <!---->
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;
